worker-core/src/main/java/com/github/workerframework/core/WorkerApplication.java

```diff
@@ -143,25 +145,26 @@ public void start() {
 
             @Override
             public void stop() {
-                LOG.info("Worker stop requested, allowing in-progress tasks to complete.");
+                LOG.info("Worker stop requested.");
+                
                 workerQueue.shutdownIncoming();
-                while (!wtp.isIdle()) {
+
+                final long startTime = System.currentTimeMillis();
+                
+                while(wtp.getBacklogSize() > 0 && System.currentTimeMillis() - startTime < SHUTDOWN_DURATION) {
```

Comment:
Are we sure about this strategy of trying to clear the backlog during shutdown?  Trying to complete in-progress tasks makes sense but if I'm understanding this right it is intentionally allowing new tasks to begin that haven't been passed to the worker code yet.  Wouldn't we be better to clear these backlog queues and let the next worker instance pick them up?

Reply:
The prefetched message have their delivery count increased and this goes towards the poison message identification. 
I take your point that starting more messages increases the chances they get terminated after the timeout.
Let's aim to talk it through at some stage.